### PR TITLE
Fixes

### LINKED ISSUE #39: Divide fn sub-cmd structs
The fn command file fn.go can be refactored. Some code could be moved to pkg perhaps and the several sub-commands could be moved in their own file.

### PR DESCRIPTION
**Pull request by mattrent**
This PR refactors the `fn` module in multiple files for each subcommand and handles the cases where a 0-byte file or a file with the wrong extension is passed to `fn create`.

This closes #39, #67.

### CODE REVIEW

File: internal/command/fn/delete.go [hunk 1]
```diff
@@ -12,4 +12,26 @@
...
+
+import (
+	"context"
+
+	"github.com/funlessdev/fl-cli/pkg/client"
+	"github.com/funlessdev/fl-cli/pkg/log"
+)
+
+type Delete struct {
+	Name      string `arg:"" name:"name" help:"name of the function to delete"`
```
Comment (giusdp) [inline internal/command/fn/delete.go:25]: When deleting a function, the name arg should be required.

File: internal/command/fn/create.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	"path"
+	"strings"
+
+	"github.com/funlessdev/fl-cli/pkg/build"
+	"github.com/funlessdev/fl-cli/pkg/client"
+	"github.com/funlessdev/fl-cli/pkg/log"
+)
+
+type Create struct {
+	Name       string `arg:"" name:"name" help:"name of the function to create"`
```
Comment (giusdp) [inline internal/command/fn/create.go:31]: Name should be also required.

File: internal/command/fn/invoke.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+	"context"
+	"encoding/json"
+	"errors"
+
+	"github.com/funlessdev/fl-cli/pkg/client"
+	"github.com/funlessdev/fl-cli/pkg/log"
+)
+
+type Invoke struct {
+	Name      string            `arg:"" name:"name" help:"name of the function to invoke"`
```
Comment (giusdp) [inline internal/command/fn/invoke.go:27]: Also in invoke the name should be required.

**Review by giusdp — CHANGES_REQUESTED** [review]

File: internal/command/fn/delete.go (hunk 1 above)
Comment (mattrent) [reply internal/command/fn/delete.go:25]: I think args in kong are required by default (https://github.com/alecthomas/kong#supported-tags); I tried running `fn delete` without a name and it returned `error: expected "<name>"`

Comment (giusdp) [reply internal/command/fn/delete.go:25]: Perfect, then we can resolve these conversations.

**Review by giusdp — APPROVED** [review]

**Pull request merged**